mlflow/models/resources.py

```diff
@@ -119,14 +118,16 @@ class _ResourceBuilder:
     def from_resources(
         resources: List[Resource], api_version: str = DEFAULT_API_VERSION
     ) -> Dict[str, Dict[ResourceType, List[Dict]]]:
-        resource_dict = defaultdict(lambda: defaultdict(list))
+        resource_dict = {}
```

Comment:
any reasons for doing this over default dict?

Reply:
Mostly a style thing for us on the OSS side. The general pros/cons for defaultdict vs. setdefault don't really apply for this use case (we're not doing frequent mutations to this structure and the read-side performance is identical between the two, so there's none of the downsides for handling the access of a value and potentially creating a new key on read that has a None type value). 
We generally lean on the side of standard library operations like this for readability for the OSS community :)